GitCommands/RevisionReader.cs

```diff
@@ -162,8 +162,8 @@ ArgumentBuilder BuildArguments()
                 {
                     "-z",
                     $"--pretty=format:\"{fullFormat}\"",
-                    { AppSettings.OrderRevisionByDate, "--date-order", "--topo-order" },
-                    { AppSettings.ShowReflogReferences, "--reflog" },
+                    { AppSettings.OrderRevisionByDate, "--date-order" },
+                    { AppSettings.ShowReflogReferences, "--reflog --topo-order" }, // if reflog is used, the revisions are not returned in topo-order. Force topo order, since we require topo order for the revision graph.
```

Comment:
Should this be visible in settings, so date-order is set when reflog is set?

Reply:
Its topo order. The topo-order means that git.exe will rearrange the order of the commits in how they are returned from the git log command. Topo order forces a specific order in parent traversal. Not sure how this exaclty works. Normally git log will return all commits in child->parent order. Adding topo-order also orderes between branches. This can be usefull.

In case of --reflog, the log is used to determine the order of the commits to be returned. When you rebase often, the log is not always in the order of child->parent. This is a problem because our graph algorithm requres a child->parent order.

So to be short; not sure what we should make visible in the settings. I do agree that for advanced users it is helpfull to know what a setting does exactly. I think the same applies about settings that slow down the applications. For example, reflog and date-order slow down the application in some cases. Shouldn't we warn the user is some way?

Reply:
We should have a wikipage with performance suggestions and have reasonable defaults
(maybe a link from settings)

A gold plated solution could be a popup if an operation takes too long time, I vote no